src/segmentedfilesystem/PacketManager.java

```diff
@@ -0,0 +1,88 @@
+package segmentedfilesystem;
+
+import java.io.IOException;
+import java.util.*;
+import java.net.DatagramPacket;
+
+public class PacketManager {
+    ReceivedFile[] files;
+
+    public PacketManager() {
+        files = new ReceivedFile[3];
+    }
+
+    public void intake(DatagramPacket receivedPacket) {
+        byte[] data;
+        data = receivedPacket.getData();
+        Packet packet = constructPacket(data, receivedPacket.getLength());
+        // know we'll only receive 3 files, hard code files list to 3
+        // boolean foundFile = false;
+        for (int i = 0; i < 3; i++) {
+            if (files[i] == null) {
+                ReceivedFile newFile = new ReceivedFile(packet.getFileID(), packet);
+		files[i] = newFile;
+                break;
+            } else if (files[i].getFileID() == packet.getFileID()) {
+                files[i].addPacket(packet);
+                break;
+            }
+        }
+    }
+
+    public boolean allPacketsReceived() {
+        int allComplete = 0;
+        for (int i = 0; i < 3; i++) {
+            if (files[i] == null) {
+                return false;
+            } else if (files[i].isCompleted()) {
+                allComplete++;
+            }
+        }
+
+        if (allComplete == 3) {
+            return true;
+        }
+	
+        return false;
+    }
+
+    public Packet constructPacket(byte[] data, int length){
```

Comment:
This is a nice example of a factory method. You could have made it static and moved it into the Packet class, which would help move the packet logic into that class hierarchy.